### PR TITLE
Use FindCBLAS if ENABLE_BLAS=ON.

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
The real condition to use BLAS in Lightning is the presence of `cblas.h`. This is not detected properly by our current CMake file system. This is currently a blocker in Conda-Forge after some updates to their build env. 

**Description of the Change:**
Mimic Jet's way of finding CBLAS. From now on, if `ENABLE_BLAS=ON` and CBLAS is not found, the building will stop, instead of silently fallback to not using BLAS.

**Benefits:**
Unblock Conda-Forge migration PR. Make sure that `ENABLE_BLAS=ON` really implies using BLAS.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 ### Improvements
 
+* Detect CBLAS if `ENABLE_BLAS=ON` making sure that BLAS is linked as expected.
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:15]:
```suggestion
* Detect MKL or CBLAS if `ENABLE_BLAS=ON` making sure that BLAS is linked as expected.
```

File: cmake/FindCBLAS.cmake [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# ==================================================================================================
```
Comment (mlxd) [inline cmake/FindCBLAS.cmake:1]: Maybe worth adding an ack to here https://github.com/PennyLaneAI/pennylane-lightning/#acknowledgements

File: cmake/FindMKL.cmake [hunk 3]
```diff
@@ -0,0 +1,78 @@
+###########################################################
```
Comment (mlxd) [inline cmake/FindMKL.cmake:1]: Same as above.

**Review by mlxd — COMMENTED** [review]
Thanks @vincentmr .
I assume the new files will also respect the CMake FindBLAS `BLA_VENDOR` arg if we aim to use that to request a given BLAS provider?

If both NETLIB and OpenBLAS are installed, I recall this being an issue where the NETLIB version was pulled rather than OpenBLAS atomically.

File: cmake/FindCBLAS.cmake (hunk 2 above)
Comment (AmintorDusko) [reply cmake/FindCBLAS.cmake:1]: I guess he meant that you should consider adding the acknowledgment to these contributions to the respective section on the readme file.

File: cmake/FindMKL.cmake (hunk 3 above)
Comment (AmintorDusko) [reply cmake/FindMKL.cmake:1]: Same as above :wink:.

**Comment by vincentmr** [comment]
This PR is related to [PR 33](https://github.com/conda-forge/pennylane-feedstock/pull/33). The current patch fixes issues finding CBLAS in Conda environments. I think this could be merged as is. @mlxd @AmintorDusko are you good with this?

File: cmake/FindCBLAS.cmake (hunk 2 above)
Comment (vincentmr) [reply cmake/FindCBLAS.cmake:1]: Forgot to update this. After speaking with @mlxd , I think the original contributor and project (Cedric Nugteren <www.cedricnugteren.nl>) are already acknowledged (or did I miss the point?).

**Review by AmintorDusko — APPROVED** [review]
Good job @vincentmr. Nothing more to add.

File: cmake/FindMKL.cmake [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+# Stage 1: find the root directory
+###########################################################
+
+set(MKLROOT_PATH $ENV{MKLROOT})
+
+if (NOT MKLROOT_PATH)
+  if (EXISTS "/opt/intel/mkl")
+    set(MKLROOT_PATH "/opt/intel/mkl")
+  endif ()
+endif ()
```
Comment (mlxd) [inline cmake/FindMKL.cmake:25]: I wonder if this now needs updating with the change in oneAPI's MKL release. Maybe we should add `/opt/intel/oneapi/mkl/latest/` or similar to the paths here, or just replace the old path with this

**Review by mlxd — COMMENTED** [review]
Thanks @vincentmr 
One thought from me on how to handle the oneAPI/mkl path, since this file was written before that change was common-place.

File: cmake/FindMKL.cmake (hunk 4 above)
Comment (vincentmr) [reply cmake/FindMKL.cmake]: Ok, made this more robust. I installed OneAPI, tested the options and everything seems in order now.

File: cmake/FindMKL.cmake [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+if (NOT MKLROOT_PATH)
+    foreach(P IN ITEMS "/opt/intel/oneapi/mkl/latest" "/opt/intel/mkl")
+        if (EXISTS "${P}")
+            set(MKLROOT_PATH "${P}")
+            message(STATUS "Found MKL: ${P}")
+            break()
+        endif ()
+  endforeach()
+endif ()
```
Comment (AmintorDusko) [inline cmake/FindMKL.cmake:33]: The indentation spacing is inconsistent. Could you please fix that?

Comment (vincentmr) [reply cmake/FindMKL.cmake]: Never let an optician say you don't have good eyes, sir!

Comment (AmintorDusko) [reply cmake/FindMKL.cmake]: Looks good now!

**Review by mlxd — APPROVED** [review]
Nice work @vincentmr

**Pull request merged**